tools/check_test.js

```diff
@@ -0,0 +1,170 @@
+    } else if (status == 'timeout') {
+      driver.logger.message('TIMEOUT : ' + filename, status);
+    }
+    driver.fIdx++;
+    driver.runNextTest();
+  });
+
+  this.logger = new Logger();
+
+  this.os = process.platform;
+  this.board = process.iotjs.board;
+
+  this.root = absolutePath(root);
+  process.chdir(this.root);
+
+  return this;
+}
+
+Driver.prototype.config = function(config) {
+  var configSrc = fs.readFileSync(absolutePath(config)).toString();
+  this.tests = eval(configSrc);
+  this.dIdx = 0;
+  this.dLength = Object.keys(this.tests).length;
+
+  this.nextTestSet();
+}
+
+Driver.prototype.getAttrs = function() {
+  var content = fs.readFileSync(absolutePath('attrs.js')).toString();
+  var attrs = eval(content);
+
+  var dirname = this.dirname();
+  if (dirname == 'run_fail') {
+    var files = this.tests[dirname];
+    for (var fIdx in files) {
+      var file = files[fIdx];
+      if (!attrs[file]) {
+        attrs[file] = {};
+      }
+      attrs[file].fail = true;
+    }
+  }
+
+  return attrs;
+};
+
+Driver.prototype.runNextTest = function() {
+  if (this.dIdx == this.dLength) {
+    this.finish();
+  } else {
+    if (this.fIdx == this.fLength) {
+      process.chdir(this.root);
+      this.dIdx++;
+      if (this.dIdx == this.dLength) {
+        this.finish();
+      } else {
+        this.nextTestSet();
+        this.runNextTest();
+      }
+    } else {
+      var filename = this.filename();
+      var attr = this.attrs[filename] || {};
+
+      if (attr.skip &&
+        (attr.skip.indexOf('all') >= 0 || attr.skip.indexOf(this.os) >= 0)) {
+          this.emitter.emit('nextTest', this, 'skip', attr);
+          return;
+      }
+
+      this.timer = null;
+      if (attr.timeout) {
+        var timeout = attr.timeout['all'] || attr.timeout[this.os];
+        if (timeout) {
+          var that = this;
+          this.timer = setTimeout(function () {
+            that.runner.finish('timeout');
+          }, timeout * 1000);
+        }
+      }
+
+      this.runner = new Runner(this);
+      this.runner.run();
+    }
+  }
+};
+
+Driver.prototype.nextTestSet = function() {
+  this.fIdx = 0;
+  var dirname = this.dirname();
+  this.fLength = this.tests[dirname].length;
+  process.chdir(absolutePath(dirname));
+  this.attrs = this.getAttrs();
+  this.logger.message("\n");
+  this.logger.message(">>>> " + dirname, "summary");
+}
+
+Driver.prototype.dirname = function() {
+  return Object.keys(this.tests)[this.dIdx]
+};
+
+Driver.prototype.filename = function() {
+  var dirname = this.dirname();
+  var filename = this.tests[dirname][this.fIdx];
+  return filename;
+}
+
+Driver.prototype.test = function() {
+  var filename = this.filename();
+  var content =  fs.readFileSync(absolutePath(filename)).toString();
+  return content;
+};
+
+Driver.prototype.finish = function() {
+  this.logger.message('finish all tests', this.logger.summary);
+}
+
+var driver = new Driver();
+
+process.exit = function () {
```

Comment:
If that's the problem, then how about using `EventEmitter.prototype.removeListener`?